### PR TITLE
Ainize

### PR DESCRIPTION
**Pull request by minsulee2**
- Adopt Ainize
- Now enable to write data on the blockchain
- Can be only tested on Ainized docker env

### CODE REVIEW

**Review by platfowner — APPROVED** [review]
LGTM
Nice job done!

File: env.js [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
 const OPENAI_API_KEY = process.env.OPENAI_API_KEY;
+const AINIZE_PRIVATE_KEY = process.env.AINIZE_PRIVATE_KEY;
 
-if (!OPENAI_API_KEY) {
-  throw new Error('OPENAI_API_KEY is not defined.');
+if (!OPENAI_API_KEY && !AINIZE_PRIVATE_KEY) {
+  throw new Error(`OPENAI_API_KEY(${OPENAI_API_KEY}), AINIZE_PRIVATE_KEY(${AINIZE_PRIVATE_KEY}) is not defined.`);
+}
+
+let REST_MODE;
```
Comment (jiyoung-an) [inline env.js:10]: question) what is `REST_MODE`?

File: handlers/middleware.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
-const { Utils } = require('./utils');
 const { JOB_TYPES } = require('../constants');
+const { ainizeAdmin } = require('../ainize');
+const { REST_MODE } = require('../env');
 
 class Middleware {
   static classifyJobType = (req, res, next) => {
     try {
-      const { jobType } = req.body;
+      const { jobType } = REST_MODE ? req.body :
```
Comment (jiyoung-an) [inline handlers/middleware.js:9]: Oh, so this is how it's applied...! thanks 🙂

File: handlers/utils.js [hunk 3]
```diff
@@ -2,6 +2,17 @@ class Utils {
   static serializeMessage(status, data) {
     return { status: status, data: data };
   }
+
+  static TrimUnnecessaryDataForResponseData(data) {
+    delete data.tools;
+    delete data.file_ids;
+    delete data.metadata;
```
Comment (jiyoung-an) [inline handlers/utils.js:9]: In the SDK, I've kept the `metadata`. How should we go about this? I'm fine with anything, but as same way.

**Review by jiyoung-an — APPROVED** [review]
LGTM, Thank you for hard works!

**Comment by jiyoung-an** [comment]
@minsulee2 @platfowner I checked that the assistant is created properly when I conducted an e2e test on this ainize container!!! 🎉 ([insight](https://testnet-insight.ainetwork.ai/database/values/apps/ainft721_0x84e8e8979e1c76ce81b5c584b23082231163483d/tokens/1/ai)) I'll make the necessary code changes in the SDK to align with upcoming PRs.

File: env.js [hunk 4]
```diff
@@ -1,11 +1,21 @@
 require('dotenv').config();
 
 const OPENAI_API_KEY = process.env.OPENAI_API_KEY;
+const AINIZE_PRIVATE_KEY = process.env.AINIZE_PRIVATE_KEY;
 
-if (!OPENAI_API_KEY) {
-  throw new Error('OPENAI_API_KEY is not defined.');
+if (!OPENAI_API_KEY && !AINIZE_PRIVATE_KEY) {
```
[the remark this answers is not in the document]
Comment (minsulee2) [reply env.js:6]: Cuz both necessary, added and op here.

File: env.js (hunk 1 above)
Comment (minsulee2) [reply env.js:10]: Yeah, as you mentioned this is a restful api mode for the local test manor.

File: handlers/utils.js (hunk 3 above)
Comment (minsulee2) [reply handlers/utils.js:9]: Indeed! That's what I have been thinking too. I will leave comment regarding this. Let's see.

**Comment by minsulee2** [comment]
Thanks for the fast reviews! Merging now and will refactor.

**Pull request merged**